### PR TITLE
Emqx ee kafka

### PR DESCRIPTION
**Pull request by qzhuyan**
emqx-ee: fully automated kafka rule engine setup
    
    - emqx-ee license
    - kafka create test topic t1
    - create kafka rule engine resource
    - create rule forward msg to kafka resource
    - log archive with prefix
    - kafka helper script
    - cfn output kafka broker and zk names
    - loadgen param tweaks
    - OS: install perf tool and lttng

### CODE REVIEW

File: cdk_emqx_cluster/cdk_emqx_cluster_stack.py [hunk 1]
```diff
@@ -851,6 +885,44 @@ def setup_kafka(self):
                                      client_broker=msk.ClientBrokerEncryption.TLS_PLAINTEXT)
                                  )
 
+        if True:
```
Comment (qzhuyan) [inline cdk_emqx_cluster/cdk_emqx_cluster_stack.py:888]: I haven't think about how to manage these FIS exp.  template that relies on Kafka broker's addr.

File: ssm_docs/fis-network-latency-src.yaml [hunk 2]
```diff
@@ -0,0 +1,379 @@
+description: |-
```
Comment (qzhuyan) [inline ssm_docs/fis-network-latency-src.yaml:1]: These are copied from AWS with minor tweaks.

File: ssm_docs/fis-network-packet-loss-src.yaml [hunk 3]
```diff
@@ -0,0 +1,380 @@
+description: |-
```
Comment (qzhuyan) [inline ssm_docs/fis-network-packet-loss-src.yaml:1]: These are copied from AWS with minor tweaks.

File: bin/kafka-helper.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    echo "$@" >&2
+    exit 1
+}
+
+kafka_get_cluser_arn() {
+    local cluster=$1;
+    aws kafka list-clusters --cluster-name-filter="${cluster}-kafka" | jq -r '.ClusterInfoList[].ClusterArn'
+
+}
+
```
Comment (k32) [inline bin/kafka-helper.sh]: Nit: formatting
```suggestion
    aws kafka list-clusters --cluster-name-filter="${cluster}-kafka" | jq -r '.ClusterInfoList[].ClusterArn'
}

```

File: bin/kafka-helper.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+kafka_get_cluser_zks() {
+    local cluster=$1;
+    aws kafka list-clusters --cluster-name-filter="${cluster}-kafka" | jq -r '.ClusterInfoList[].ZookeeperConnectString'
+
+}
```
Comment (k32) [inline bin/kafka-helper.sh:17]: Nit: formatting
```suggestion
    aws kafka list-clusters --cluster-name-filter="${cluster}-kafka" | jq -r '.ClusterInfoList[].ZookeeperConnectString'

}
```

File: user_data/emqx-kafka-rule-engine.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        "keyfile": {
+          "filename": "",
+          "file": ""
+        },
+        "max_batch_bytes": "900KB",
+        "min_metadata_refresh_interval": "3s",
+        "password": "",
+        "query_api_versions": true,
+        "send_buffer": "1024KB",
+        "servers": "b-2.william-k2-kafka.x7x6m7.c3.kafka.eu-west-1.amazonaws.com:9092,b-1.william-k2-kafka.x7x6m7.c3.kafka.eu-west-1.amazonaws.com:9092",
```
Comment (k32) [inline user_data/emqx-kafka-rule-engine.json:54]: Is it an example data?

**Review by k32 — APPROVED** [review]

File: user_data/emqx-kafka-rule-engine.json (hunk 6 above)
Comment (qzhuyan) [reply user_data/emqx-kafka-rule-engine.json:54]: yes, this file is just a template and while deploying the `servers` field will be rendered with values in-use.

File: cdk_emqx_cluster/cdk_emqx_cluster_stack.py (hunk 1 above)
Comment (thalesmg) [reply cdk_emqx_cluster/cdk_emqx_cluster_stack.py:888]: Perhaps we could make the broker addresses an output of this stack, and then [import](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.core/Fn.html#aws_cdk.core.Fn.import_value) that output as in input to the Chaos app?

Something like: https://bobbyhadz.com/blog/import-value-aws-cdk-cross-stack

**Pull request merged**